homeassistant/components/media_player/kodi.py

```diff
@@ -15,20 +20,50 @@
     SUPPORT_TURN_OFF, MediaPlayerDevice, PLATFORM_SCHEMA)
 from homeassistant.const import (
     STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING, CONF_HOST, CONF_NAME,
-    CONF_PORT, CONF_USERNAME, CONF_PASSWORD)
+    CONF_PORT, CONF_USERNAME, CONF_PASSWORD, EVENT_HOMEASSISTANT_STOP)
 import homeassistant.helpers.config_validation as cv
 
-REQUIREMENTS = ['jsonrpc-requests==0.3']
+REQUIREMENTS = ['websockets==3.2']
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_TCP_PORT = 'tcp_port'
 CONF_TURN_OFF_ACTION = 'turn_off_action'
 
 DEFAULT_NAME = 'Kodi'
 DEFAULT_PORT = 8080
+DEFAULT_TCP_PORT = 9090
+DEFAULT_TIMEOUT = 5
 
 TURN_OFF_ACTION = [None, 'quit', 'hibernate', 'suspend', 'reboot', 'shutdown']
 
+APPLICATION_NOTIFICATIONS = (
+    'Application.OnVolumeChanged',
+)
+
+PLAYER_NOTIFICATIONS = (
```

Comment:
This code is starting to acquire a lot of protocol related info. I rather see this in an external lib.